### PR TITLE
feat: add backend tests and configure pytest

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR

Added unit tests for backend routes and configured pytest for asyncio. Part of MEE-44.

### What changed?

- Added new test files: `audio_test.py`, `bots_test.py`, and `meetings_test.py`.
- Updated `requirements.txt` with new dependencies for testing.
- Created `conftest.py` to configure pytest to use asyncio as the backend.

### How to test?

In the next PR in the stack I am adding instructions for how to run the tests to the README.

### Why make this change?

This change introduces unit tests for the backend routes, improving code reliability and making it easier to catch potential issues early in the development process.

This is very important as it is a requirement for our capstone, and will additionally increase confidence in the reliability of our service for potential users.

### Links

- FastAPI Async Tests Docs: https://fastapi.tiangolo.com/advanced/async-tests/#example

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/10?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/11?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/10?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/9?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @owengretzinger and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/backend/requirements.txt
[the remark this answers is not in the document]
Comment (sahibkhokhar) [reply src/backend/requirements.txt:1]: +1 for rye

[the remark this answers is not in the document]
Comment (owengretzinger) [reply src/backend/requirements.txt:1]: Sure; want me to update this PR or should I do it in the next one?

**Comment by err53** [comment]
I think a separate PR is fine

Jason Huang
e: ***@***.***
m: (647) 631-6610

**Review by err53 — APPROVED** [review]

File: src/backend/requirements.txt
[the remark this answers is not in the document]
Comment (err53) [reply src/backend/requirements.txt:1]: I think a separate PR is fine

**Pull request merged**